### ISSUE TITLE
Issue with running the example data

### ISSUE THREAD

**Issue by barosmarko** [sdparekh/zUMIs#17]
Dear zUMIs developers,

I've been trying to run zUMIs with this command line:

`bash ./zUMIs/zUMIs-master.sh -f ./zUMIs/ExampleData/barcoderead_HEK.1mio.fq.gz -r ./zUMIs/ExampleData/cDNAread_HEK.1mio.fq.gz -c 1-6 -m 7-16 -l 50 -g ./reference/hg38_chr22/ -a ./reference/GRCh38.84.chr22.gtf -n chr22test -p 8 -i /Users/markobaros/zumis_proba/zUMIs/ -o /Users/markobaros/zumis_proba/`

This is what I get when I run:

readlink: illegal option -- f
usage: readlink [-n] [file ...]
usage: dirname path
Your jobs will run on this machine.

Make sure you have more than 1.8G RAM and 8 processors available.

Your jobs will be started from filtering.

 You provided these parameters:
 SLURM workload manager:	no
 Summary Stats to produce:	yes
 Start the pipeline from:	filtering
 A custom mapped BAM:		NA
 Custom filtered FASTQ:		no
 Barcode read:			./zUMIs/ExampleData/barcoderead_HEK.1mio.fq.gz
 cDNA read:			./zUMIs/ExampleData/cDNAread_HEK.1mio.fq.gz
 Study/sample name:		chr22test
 Output directory:		/Users/markobaros/zumis_proba/
 Cell/sample barcode range:	1-6
 UMI barcode range:		7-16
 Retain cell with >=N reads:	100
 Genome directory:		./reference/hg38_chr22/
 GTF annotation file:		./reference/GRCh38.84.chr22.gtf
 Number of processors:		8
 Read length:			50
 Strandedness:			0
 Cell barcode Phred:		20
 UMI barcode Phred:		20
 # bases below phred in CellBC:	1
 # bases below phred in UMI:	1
 Barcodes:			NA
 zUMIs directory:		/Users/markobaros/zumis_proba/zUMIs/
 STAR executable		STAR
 samtools executable		samtools
 pigz executable		pigz
 Rscript executable		Rscript
 Additional STAR parameters:
 STRT-seq data:			no
 InDrops data:			no
 Library read for InDrops:	NA
 Barcode read2(STRT-seq):	NA
 Barcode read2 range(STRT-seq):	0-0
 Bases(G) to trim(STRT-seq):	3
 Subsampling reads:		0

du: illegal option -- B
usage: du [-H | -L | -P] [-a | -s | -d depth] [-c] [-h | -k | -m | -g] [-x] [-I mask] [file ...]
Raw reads: 1000000
Filtered reads: 853296

Jan 08 15:58:17 ..... started STAR run
Jan 08 15:58:17 ..... loading genome
Jan 08 15:58:21 ..... processing annotations GTF
Jan 08 15:58:21 ..... inserting junctions into the genome indices
Jan 08 15:58:31 ..... started 1st pass mapping
Jan 08 15:59:49 ..... finished 1st pass mapping
Jan 08 15:59:49 ..... inserting junctions into the genome indices
Jan 08 15:59:58 ..... started mapping
Jan 08 16:01:20 ..... finished successfully
[bam_sort_core] merging from 0 files and 8 in-memory blocks...
[bam_sort_core] merging from 0 files and 8 in-memory blocks...
Loading required package: optparse
[1] "I am loading useful packages..."
[1] "2018-01-08 16:01:24 CET"
[1] "I am loading Rsubread 1.26.0..."
[1] "I am making annotations in SAF... This will take less than 3 minutes..."
[1] "2018-01-08 16:01:35 CET"
Import genomic features from the file as a GRanges object ... OK
Prepare the 'metadata' data frame ... OK
Make the TxDb object ... OK
There were 17 warnings (use warnings() to see them)
'select()' returned 1:many mapping between keys and columns
[1] "I am making count tables...This will take a while!!"
[1] "2018-01-08 16:01:40 CET"

        ==========     _____ _    _ ____  _____  ______          _____
        =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \
          =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |
            ====      \___ \| |  | |  _ <|  _  /|  __|   / /\ \ | |  | |
              ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |
        ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/
       Rsubread 1.26.0

//========================== featureCounts setting ===========================\\
||                                                                            ||
||             Input files : 1 BAM file                                       ||
||                           S /Users/markobaros/zumis_proba//chr22test.a ... ||
||                                                                            ||
||      Dir for temp files : .                                                ||
||                 Threads : 1                                                ||
||                   Level : meta-feature level                               ||
||              Paired-end : no                                               ||
||         Strand specific : no                                               ||
||      Multimapping reads : primary only                                     ||
|| Multi-overlapping reads : not counted                                      ||
||   Min overlapping bases : 1                                                ||
||                                                                            ||
\\===================== http://subread.sourceforge.net/ ======================//

//================================= Running ==================================\\
||                                                                            ||
|| Load annotation file ./.Rsubread_UserProvidedAnnotation_pid53259 ...       ||
||    Features : 5737                                                         ||
||    Meta-features : 698                                                     ||
||    Chromosomes/contigs : 3                                                 ||
||                                                                            ||
|| Process BAM file /Users/markobaros/zumis_proba//chr22test.aligned.sort ... ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 853296                                                    ||
||    Successfully assigned reads : 46553 (5.5%)                              ||
||    Running time : 0.04 minutes                                             ||
||                                                                            ||
||                         Read assignment finished.                          ||
||                                                                            ||
\\===================== http://subread.sourceforge.net/ ======================//

        ==========     _____ _    _ ____  _____  ______          _____
        =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \
          =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |
            ====      \___ \| |  | |  _ <|  _  /|  __|   / /\ \ | |  | |
              ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |
        ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/
       Rsubread 1.26.0

//========================== featureCounts setting ===========================\\
||                                                                            ||
||             Input files : 1 BAM file                                       ||
||                           S /Users/markobaros/zumis_proba//chr22test.a ... ||
||                                                                            ||
||      Dir for temp files : .                                                ||
||                 Threads : 1                                                ||
||                   Level : meta-feature level                               ||
||              Paired-end : no                                               ||
||         Strand specific : no                                               ||
||      Multimapping reads : primary only                                     ||
|| Multi-overlapping reads : not counted                                      ||
||   Min overlapping bases : 1                                                ||
||                                                                            ||
\\===================== http://subread.sourceforge.net/ ======================//

//================================= Running ==================================\\
||                                                                            ||
|| Load annotation file ./.Rsubread_UserProvidedAnnotation_pid53259 ...       ||
||    Features : 26131                                                        ||
||    Meta-features : 1190                                                    ||
||    Chromosomes/contigs : 4                                                 ||
||                                                                            ||
|| Process BAM file /Users/markobaros/zumis_proba//chr22test.aligned.sort ... ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 853296                                                    ||
||    Successfully assigned reads : 21890 (2.6%)                              ||
||    Running time : 0.04 minutes                                             ||
||                                                                            ||
||                         Read assignment finished.                          ||
||                                                                            ||
\\===================== http://subread.sourceforge.net/ ======================//

**Error in if (sigmasq > signif(.Machine$double.xmax, 6)) { :
  missing value where TRUE/FALSE needed
Calls: makeGEprofile ... eval -> mclustBIC -> mstep -> eval -> eval -> mstepE
Execution halted
[1] "I am loading useful packages for plotting..."
[1] "2018-01-08 16:01:54 CET"
Error in gzfile(file, "rb") : cannot open the connection
Calls: readRDS -> gzfile
In addition: Warning message:
In gzfile(file, "rb") :
  cannot open compressed file '/Users/markobaros/zumis_proba//zUMIs_output/expression/chr22test.dgecounts.rds', probable reason 'No such file or directory'
Execution halted**

I've tried to trace the problem, and I've found that it comes from zUMIs-dge.R:

```
210 BCselection <- function(fullstats){
211 | tmp<-mclust::Mclust(log10(fullstats$nreads), modelNames = c("E","V"))
```
And finally problem comes from mclust.R and it seems that problem is that maximization step can't be computed, but I don't know why. This probably affects zUMIs-dge.R at line 435 and it doesn't have anything to save:

`saveRDS(AllCounts,file=paste(out,"/zUMIs_output/expression/",sn,".dgecounts.rds",sep=""))`

If it helps, here is the chr22.Log.final.out:
                                 Started job on |	Jan 08 15:58:17
                             Started mapping on |	Jan 08 15:59:58
                                    Finished on |	Jan 08 16:01:20
       Mapping speed, Million of reads per hour |	37.46

                          Number of input reads |	853296
                      Average input read length |	50
                                    UNIQUE READS:
                   Uniquely mapped reads number |	130719
                        Uniquely mapped reads % |	15.32%
                          Average mapped length |	48.41
                       Number of splices: Total |	8929
            Number of splices: Annotated (sjdb) |	4497
                       Number of splices: GT/AG |	8244
                       Number of splices: GC/AG |	185
                       Number of splices: AT/AC |	3
               Number of splices: Non-canonical |	497
                      Mismatch rate per base, % |	5.04%
                         Deletion rate per base |	0.01%
                        Deletion average length |	1.42
                        Insertion rate per base |	0.10%
                       Insertion average length |	2.07
                             MULTI-MAPPING READS:
        Number of reads mapped to multiple loci |	78370
             % of reads mapped to multiple loci |	9.18%
        Number of reads mapped to too many loci |	83
             % of reads mapped to too many loci |	0.01%
                                  UNMAPPED READS:
       % of reads unmapped: too many mismatches |	0.00%
                 % of reads unmapped: too short |	75.43%
                     % of reads unmapped: other |	0.06%
                                  CHIMERIC READS:
                       Number of chimeric reads |	0
                            % of chimeric reads |	0.00%

And when I run `ls -altr zUMIs_output/` i get this:

drwxr-xr-x   2 markobaros  staff   68 29 Dec 14:04 stats
drwxr-xr-x   5 markobaros  staff  170 29 Dec 14:04 .
drwxr-xr-x   3 markobaros  staff  102 29 Dec 14:08 expression
drwxr-xr-x   4 markobaros  staff  136  8 Jan 16:02 filtered_fastq
drwxr-xr-x  16 markobaros  staff  544  8 Jan 16:02 ..

Hope this can lead you to the problem that I can't see!

Thank you in advance!

Best,
Marko

**Comment by sdparekh** [comment]
Hi Marko,

Thanks for using zUMIs. Sorry that it took sometime to answer but I could not reproduce your error. 

I don't think it is related to this error but I would try the following just in case:
1) give absolute paths to all files and folders
2) clone zUMIs again if you are not using the latest commit
3) send me your R sessionInfo()
4) can you also send me the md5sums of the input files

Please feel free to post again if the issue persists. 

Best,
Swati

**Comment by barosmarko** [comment]
Hey Swati,

No problem, you were swift!

It's working right now! I've updated my R to 3.4.3 and it has done
everything with no problems at all! The obtained plots are as same as the
ones in the /ExampleData/zUMIs_output/stats/, but most of the output files
are slightly different, but I think it is nothing special in terms of
differences, isn't it?

These results I get by running on macOS, but I've had to change 'zcat' in
codes to 'gunzip -c', since in some OS zcat expect that file has the
*.Z* extension
e.g. *some_file.tar.gz**.Z*, and the task can fail. Probably, most of your
users are using ubuntu for running, and I will do it as well, but if
someone wants to run it on different OS will face this problem.

My next step is to run it with some 10x data on the EC2 instance. If I see
something strange, I'll let you know.

All kudos for your zUMIs work and for your responsiveness!

Best,
Marko
